.github/workflows/autolabel-pullrequests.yml

```diff
@@ -41,15 +41,36 @@ jobs:
         with:
           script: |
 
-            let lbl;
-            let lbls = new Set(); // Set of label strings
+            let newCompLbls = new Set(); // Set of new label strings
 
+            const curLblObjs = await github.issues.listLabelsOnIssue({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            // Delete current component labels from PR
+            // Will identify new component labels later based on the files modified in the PR
+            for (const l of curLblObjs['data']) {
+              //console.log("Current LabelName: " + l['name'] + " LabelDescription: " + l['description']);
+              if(l['name'].startsWith("component: "))  {
+                github.issues.removeLabel({
```

Comment:
you should not remove all labels everytime. You should only apply the difference.

If there is one less remove one less, if there is one more just add one more.

Don't delete every components labels before applying them.

Reply:
@quentinDERORY - I understand and will do the needful. I should have it ready asap.

Reply:
@quentinDERORY the changes are done and tested as per your last review comments. Please review and let me know if you have any questions. Thank you !